### PR TITLE
[SC-437] Fixing coverage by adding unit tests

### PR DESCRIPTION
**Pull request by hexonaut**
Adding unit tests to the receiver contracts for more robust testing and it fixes coverage issues.

Please note I also updated the import to relative to fix an issue with having this as a library. It wouldn't resolve the types properly otherwise.

Also, fixing a bug with subsequent message relays on all testing frameworks.

### CODE REVIEW

File: test/mocks/TargetContractMock.sol [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.8.0;
+
+contract TargetContractMock {
+
+    uint256 public data;
```
Comment (barrutko) [inline test/mocks/TargetContractMock.sol:6]: I'd rename `uint256 public data;` to `counter` or `callCount`.

It'd make tests like this cleaner
```
        assertEq(target.callCount(), 0);
        vm.prank(address(l2CrossDomain));
        TargetContractMock(address(receiver)).someFunc();
        assertEq(target.callCount(), 1);
```

File: test/mocks/TargetContractMock.sol [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    function someFunc() external {
+        data++;
+    }
+
+    function revertFunc() external pure {
+        revert("error");
```
Comment (barrutko) [inline test/mocks/TargetContractMock.sol:13]: I'd change the revert string to sth like `TargetContract/error`. Again it's make it more clear in tests that this is the error coming from the target contract.
```
        vm.prank(address(l2CrossDomain));
        vm.expectRevert("TargetContract/error");
        TargetContractMock(address(receiver)).revertFunc();
```

**Review by barrutko — CHANGES_REQUESTED** [review]
Two small suggestions, other that that looks good.

File: test/mocks/TargetContractMock.sol (hunk 1 above)
Comment (lucas-manuel) [reply test/mocks/TargetContractMock.sol:6]: Agree

File: test/mocks/TargetContractMock.sol (hunk 2 above)
Comment (lucas-manuel) [reply test/mocks/TargetContractMock.sol:13]: Agree

File: test/CCTPReceiver.t.sol [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            sourceDomainId,
+            bytes32(uint256(uint160(randomAddress))),
+            abi.encodeCall(TargetContractMock.someFunc, ())
+        );
+    }
+
+    function test_handleReceiveMessage_success() public {
+        assertEq(target.data(), 0);
+        vm.prank(destinationMessenger);
+        receiver.handleReceiveMessage(
```
Comment (lucas-manuel) [inline test/CCTPReceiver.t.sol:79]: May as well assert return value here

File: test/IntegrationBase.t.sol [hunk 4]
```diff
@@ -105,6 +105,16 @@ abstract contract IntegrationBaseTest is Test {
         assertEq(moSource.length(), 2);
         assertEq(moSource.messages(0), 3);
         assertEq(moSource.messages(1), 4);
+
+        // One more message to destination
```
Comment (lucas-manuel) [inline test/IntegrationBase.t.sol:109]: Don't we want another to source as well to make sure that both relays handle correctly?

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: test/mocks/TargetContractMock.sol (hunk 1 above)
Comment (hexonaut) [reply test/mocks/TargetContractMock.sol:6]: fixed

File: test/mocks/TargetContractMock.sol (hunk 2 above)
Comment (hexonaut) [reply test/mocks/TargetContractMock.sol:13]: changed

File: test/CCTPReceiver.t.sol (hunk 3 above)
Comment (hexonaut) [reply test/CCTPReceiver.t.sol:79]: added

File: test/IntegrationBase.t.sol (hunk 4 above)
Comment (hexonaut) [reply test/IntegrationBase.t.sol:109]: ya good point, added

File: test/CCTPReceiver.t.sol [hunk 5]
```diff
@@ -69,24 +69,25 @@ contract CCTPReceiverTest is Test {
...
-        receiver.handleReceiveMessage(
+        bool result = receiver.handleReceiveMessage(
             sourceDomainId,
             sourceAuthority,
-            abi.encodeCall(TargetContractMock.someFunc, ())
+            abi.encodeCall(TargetContractMock.increment, ())
         );
-        assertEq(target.data(), 1);
+        assertEq(result, true);
+        assertEq(target.count(), 1);
```
Comment (lucas-manuel) [inline test/CCTPReceiver.t.sol:85]: Align

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: test/CCTPReceiver.t.sol (hunk 5 above)
Comment (hexonaut) [reply test/CCTPReceiver.t.sol]: done

**Review by barrutko — APPROVED** [review]
LGTM

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**